components/microsoft_onedrive/actions/get-table/get-table.mjs

```diff
@@ -0,0 +1,92 @@
+import axios from "@pipedream/platform";
+
+export default {
+  name: "Get Table",
+  description: "Retrieve a table from an Excel spreadsheet stored in OneDrive",
+  key: "onedrive-get-excel-table",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    microsoft_onedrive: {
+      type: "app",
+      app: "microsoft_onedrive",
+      label: "",
+      description: "",
+    },
+    itemId: {
+      type: "string",
+      label: "Spreadsheet",
+      description: "Search for the file by name, only xlsx files are supported",
+      useQuery: true,
+      async options( context ) {
+        const response = await axios(context, {
+          url: `https://graph.microsoft.com/v1.0/me/drive/search(q='${context?.query ?? ""} .xlsx')?select=name,id`,
+          headers: {
+            Authorization: `Bearer ${this.microsoft_onedrive.$auth.oauth_access_token}`,
+          },
+        });
+        const filtered = response.value.filter(
+          ( { name } ) => { return name.substring(name.length - 4) == "xlsx"; },
+        );
+        return filtered.map( function ({
+          name, id,
+        }) {
+          return ({
+            label: name,
+            value: id,
+          });
+        });

```

Comment:
**Minor**
Simplification:

```suggestion
        return response.value
          .filter(({ name }) => name.endsWith(".xlsx"))
          .map(({
            name, id,
          }) => ({
            label: name,
            value: id,
          }));
```

Reply:
Doh. Yes that should be chained.

I was considering refactoring the return into a function that accepts a string for the file extension, but decided against it and forgot to clean up.